File: Binary Search/O(nlogn)_time_and_space.js
function binarySearch(array, target) {
  // Write your code here.
  return findEle(array, 0, array.length - 1, target);
}

const findEle = (array, start, end, target) => {
  if (start > end) return -1;
  const middleInd = Math.floor((end + start) / 2);
  if (array[middleInd] === target) return middleInd;
  else if (array[middleInd] > target)
    return findEle(array, 0, middleInd - 1, target);
  else return findEle(array, middleInd + 1, end, target);
};

File: Searching/Binary Search/O(nlogn)_time_and_space.py
def binarySearch(array, target):
    # Write your code here.
    return findTerm(array,target,0,len(array)-1)

def findTerm(array,target,start,end):
    if end<start: return -1
    midInd=(start+end)//2
    if array[midInd] == target: return midInd
    elif array[midInd] > target: return findTerm(array,target,start,midInd-1)
    else: return findTerm(array,target,midInd+1,end)